src/main/scala/path/Path.scala

```diff
@@ -5,33 +5,41 @@ import com.originate.scalypher.action.ReturnAll
 import com.originate.scalypher.action.ReturnDistinct
 import com.originate.scalypher.action.ReturnReference
 import com.originate.scalypher.Query
+import com.originate.scalypher.MatchQuery
+import com.originate.scalypher.MergeQuery
+import com.originate.scalypher.CreateQuery
 import com.originate.scalypher.types.Referenceable
 import com.originate.scalypher.types.ReferenceableMap
-import com.originate.scalypher.where.ReferenceType
+import com.originate.scalypher.where.Reference
 import com.originate.scalypher.where.Where
 
 case class PathWithWhere(path: Path, where: Where) {
```

Comment:
not sure if you want to deal with it, but properties can also be referenced:
`match n return n.name`